eslint.config.js

```diff
@@ -0,0 +1,422 @@
+const importPlugin = require('eslint-plugin-import');
+const jest = require('eslint-plugin-jest');
+const simpleImportSort = require('eslint-plugin-simple-import-sort');
+const unicorn = require('eslint-plugin-unicorn');
+const globals = require('globals');
+const { FlatCompat } = require('@eslint/eslintrc');
+const js = require('@eslint/js');
+
+const eslintrc = new FlatCompat({
+  baseDirectory: __dirname,
+  recommendedConfig: js.configs.recommended,
+});
+
+module.exports = [
+  {
+    ignores: [
+      'node_modules',
+      'dist',
+      'jest.config.js',
+      'fixtures',
+      'coverage',
+      '__snapshots__',
+      '.docusaurus',
+      'build',
+
+      'packages/eslint-plugin-tslint/tests',
+
+      'packages/website/**/*.js',
+      'packages/website/**/*.d.ts',
+      'packages/website-eslint/**/*.js',
+      'packages/website-eslint/**/*.d.ts',
+
+      // Files copied as part of the build
+      'packages/types/src/generated/**/*.ts',
+    ],
+  },
+  js.configs.recommended,
+  eslintrc.extends([
+    'plugin:eslint-plugin/recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:@typescript-eslint/recommended-requiring-type-checking',
+  ]),
+  {
+    plugins: {
+      '@typescript-eslint': typescriptESLint,
+      '@typescript-eslint/internal': typeScriptESLintInternal,
+      deprecation,
+      'eslint-comments': eslintComments,
+      'eslint-plugin': eslintPlugin,
+      import: importPlugin,
+      jest,
+      'simple-import-sort': simpleImportSort,
+      unicorn,
+    },
+    languageOptions: {
+      sourceType: 'module',
+      globals: {
+        ...globals.node,
+      },
+      parserOptions: {
+        project: [
+          './tsconfig.eslint.json',
+          './packages/*/tsconfig.json',
+          './tests/integration/tsconfig.json',
+          /**
+           * We are currently in the process of transitioning to nx's out of the box structure and
+           * so need to manually specify converted packages' tsconfig.build.json and tsconfig.spec.json
+           * files here for now in addition to the tsconfig.json glob pattern.
+           *
+           * TODO(#4665): Clean this up once all packages have been transitioned.
+           */
+          './packages/scope-manager/tsconfig.build.json',
+          './packages/scope-manager/tsconfig.spec.json',
+        ],
+        allowAutomaticSingleRunInference: true,
+        tsconfigRootDir: __dirname,
+        warnOnUnsupportedTypeScriptVersion: false,
+        EXPERIMENTAL_useSourceOfProjectReferenceRedirect: false,
+        cacheLifetime: {
+          // we pretty well never create/change tsconfig structure - so need to ever evict the cache
+          // in the rare case that we do - just need to manually restart their IDE.
+          glob: 'Infinity',
+        },
+      },
+    },
+    rules: {
+      // make sure we're not leveraging any deprecated APIs
+      'deprecation/deprecation': 'error',
+
+      //
+      // our plugin :D
```

Comment:
[Praise] Thank you for keeping the comments. This one has always made me smile 😄